src/ray/core_worker/task_manager.h

```diff
@@ -96,8 +98,21 @@ class TaskManager : public TaskFinisherInterface {
   /// Return the spec for a pending task.
   TaskSpecification GetTaskSpec(const TaskID &task_id) const;
 
+  void SetActorCounter(TaskSpecBuilder &builder, const ActorID &actor_id);
+
+  void SetRemainingRetries(const TaskID &task_id, uint64_t retries);
+
   /// Return the number of pending tasks.
-  int NumPendingTasks() const { return pending_tasks_.size(); }
+  int NumPendingTasks() const {
+    absl::MutexLock lock(&mu_);
+    return pending_tasks_.size();
```

Comment:
This is unrelated to the change, but it looks like this method has been unprotected 🤷‍♂

Reply:
Whew thanks!